src/test/java/com/github/tomakehurst/wiremock/matching/RequestPatternTest.java

```diff
@@ -252,6 +256,70 @@ public void bindsToJsonCompatibleWithOriginalRequestPatternWithQueryParams() thr
         true);
   }
 
+  @Test
+  public void matchesExactlyWith0DistanceWhenAllRequiredFormParametersMatch() {
+    RequestPattern requestPattern =
+        newRequestPattern(PUT, urlPathEqualTo("/my/url"))
+            .withFormParam("key1", equalTo("value1"))
+            .withFormParam("key2", equalTo("value2"))
+            .build();
+
+    Map<String, FormParameter> formParameters = new HashMap<>();
+    formParameters.put("key1", new FormParameter("key1", List.of("value1")));
+    formParameters.put("key2", new FormParameter("key1", List.of("value2")));
+    MatchResult matchResult =
+        requestPattern.match(
+            mockRequest().method(PUT).url("/my/url").formParameters(formParameters));
+    assertThat(matchResult.getDistance(), is(0.0));
+    assertTrue(matchResult.isExactMatch());
+  }
+
+  @Test
+  public void returnsNon0DistanceWhenRequiredFormParameterMatchDoesNotMatch() {
+    RequestPattern requestPattern =
+        newRequestPattern(PUT, urlPathEqualTo("/my/url"))
+            .withFormParam("key1", equalTo("value1"))
+            .withFormParam("key2", equalTo("value2"))
+            .build();
+
+    Map<String, FormParameter> formParameters = new HashMap<>();
+    formParameters.put("key1", new FormParameter("key1", List.of("value555")));
+    formParameters.put("key2", new FormParameter("key1", List.of("value78")));
+
+    MatchResult matchResult =
+        requestPattern.match(
+            mockRequest().method(PUT).url("/my/url").formParameters(formParameters));
+    assertThat(matchResult.getDistance(), greaterThan(0.0));
+    assertFalse(matchResult.isExactMatch());
+  }
+
+  @Test
+  public void bindsToJsonCompatibleWithOriginalRequestPatternWithFormParams() throws Exception {
+    RequestPattern requestPattern =
+        newRequestPattern(POST, urlPathEqualTo("/my/url"))
+            .withFormParam("key1", equalTo("value1"))
+            .withFormParam("key2", matching("value2"))
+            .build();
+
+    String actualJson = Json.write(requestPattern);
+
+    JSONAssert.assertEquals(
+        "{                              \n"
+            + "    \"method\": \"GET\",       \n"
+            + "    \"urlPath\": \"/my/url\",  \n"
+            + "    \"formParameters\": {     \n"
+            + "        \"key1\": {          \n"
+            + "            \"equalTo\": \"value1\" \n"
+            + "        },                     \n"
+            + "        \"key2\": {          \n"
+            + "            \"matches\": \"value2\" \n"
+            + "        }                      \n"
+            + "    }                          \n"
+            + "}",
+        actualJson,
+        true);
+  }
+
```

Comment:
Please can we also have a test case showing that serialisation works correctly?

Reply:
sure, incorporated

Reply:
I believe you mean deserialization. I have now both tests serialization and deserialization.